modules/openapi-generator/src/test/resources/3_0/oneoOfDiscriminator.yaml

```diff
@@ -0,0 +1,201 @@
+openapi: 3.0.1
+info:
+   title: fruity
+   version: 0.0.1
+paths:
+   /:
+      get:
+         responses:
+            '200':
+               description: desc
+               content:
+                  application/json:
+                     schema:
+                        $ref: '#/components/schemas/FruitAllOfDisc'
+components:
+   schemas:
+    FruitType:
+      properties:
+        fruitType:
+          type: string
+      required:
+      - fruitType
+    FruitInlineDisc:
+      oneOf:
+        - type: object
+          required:
+            - seeds
+            - fruitType
+          properties:
+            seeds:
+              type: integer
+            fruitType:
+              type: string
+        - type: object
+          required:
+            - length
+            - fruitType
+          properties:
+            length:
+              type: integer
+            fruitType:
+              type: string
+      discriminator:
```

Comment:
From https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#discriminator-object

> When using the discriminator, inline schemas will not be considered.

Reply:
So our defualtcodegen extracts these inline schemas into separate schemas with the inline model resolver.
I'm not sure how the generator will be bale to tell that this was an inline schema because when I am processing it, it is no longer inline.
Do we have a boolean flag that indicates that a schema came from inline definition?

Reply:
Hm. So InlineModelResolver is trying to be smart about it and find a similar looking model, then using that... We'll need to change that to be spec compliant for composed schemas.

Reply:
The function has been updated to now throw an exception if a null ref is seen.
We also have a test of it. In my opinion we should handle fixing the inline model resolver for composed schemas in a separate PR as that is scope creep. The scope here is defining the discriminator map.

Reply:
@jimschubert please let me know if your concerns have been addressed.
A separate ticket has been filed to fix the inline model resolver [here](https://github.com/OpenAPITools/openapi-generator/issues/5230)

Reply:
I have one more concern. I'll post it in PR comment rather than review comment.